src/pybind/mgr/cephadm/module.py

```diff
@@ -3139,6 +3139,40 @@ def cert_store_cert_ls(self) -> Dict[str, Any]:
     def cert_store_key_ls(self) -> Dict[str, Any]:
         return self.cert_key_store.key_ls()
 
+    @handle_orch_error
+    def cert_store_get_cert(
+        self,
+        entity: str,
+        service_name: Optional[str] = None,
+        hostname: Optional[str] = None
+    ) -> str:
+        cert = self.cert_key_store.get_cert(entity, service_name or '', hostname or '')
+        if not cert:
+            err_msg = f'No cert found for entity {entity}'
+            if service_name:
+                err_msg += f' with service name {service_name}'
+            if hostname:
+                err_msg += f' with hostname {hostname}'
+            raise OrchestratorError(err_msg)
```

Comment:
No big deal, but maybe you can encapsulate this logic within a derived exception that does all the formatting for you:
```suggestion
        if not cert:
           raise OrchCertNotFoundError(entity=entity, service_name=service_name, hostname=hostname)
```
In fact, instead of capturing the return code (`cert is None`), the `get_cert()` could directly raise that exception:
```py
    @handle_orch_error
    def cert_store_get_cert(
        self,
        entity: str,
        service_name: Optional[str] = None,
        hostname: Optional[str] = None
    ) -> str:
        return self.cert_key_store.get_cert(entity, service_name or '', hostname or '')  # raises OrchCertNotFoundError
```

Reply:
I like the idea of the new exception type to do the formatting. I think I'll stick to having the exception be raised directing in `cert_store_get_cert` for now, because making the underlying cert_key_store object raise it would require changed in a handful of other spots that expect it to return an empty string if the cert isn't found and I'd like that to be kept separate from the feature being added here. Another thing I'll think about changing for the followup mentioned in https://github.com/ceph/ceph/pull/57672#discussion_r1669158004